lib/src/config_provider/path_finder.dart

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Utils for finding header paths on system.
+
+import 'dart:io';
+
+import 'package:logging/logging.dart';
+import 'package:path/path.dart' as p;
+
+final _logger = Logger('ffigen.config_provider.path_finder');
+
+/// This will return include path from either LLVM, XCode or CommandLineTools.
+List<String> getHeadersForMac() {
+  final includePath = <String>[];
+
+  /// Add system headers.
+  const systemHeaders =
+      '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include';
+  if (Directory(systemHeaders).existsSync()) {
+    _logger.fine('Added $systemHeaders to compiler-opts.');
+    includePath.add('-I' + systemHeaders);
```

Comment:
These system headers partially overlap with the C standard library (e.g. no stdarg.h, but yes stddef.h), is there any risk in including both?

Maybe system headers should be under a different key in the config yaml and be not included by default?

Reply:
> is there any risk in including both?

I am actually seeing some clash between CommandLineTools and System headers. If I add CommandLineTools to include path first and then add System headers, then I am seeing errors like `int32_t` not defined, `size_t` not defined.
My guess is that CommandLine Tools doesn't define these and set the include guard macro?

So we definitely need to add the system headers to path first incase we added commandlinetools.

Reply:
Also, CPATH is looked at after the compiler opt paths, so the user would need to specify system headers in compiler opts in case the C standard library we used was from CommandLineTools.

Reply:
Thanks for investigating!

This PR works on my side. 👍